webapp/src/js/controllers/contacts.js

```diff
@@ -253,13 +253,19 @@ var _ = require('underscore'),
         });
     };
 
-    $scope.$on('ClearSelected', function() {
+    var clearSelection = function() {
       $scope.selected = null;
       LiveList.contacts.clearSelected();
       LiveList['contact-search'].clearSelected();
+    };
+
+    $scope.$on('ClearSelected', function() {
+      clearSelection();
     });
 
     $scope.search = function() {
+      $state.go('contacts', null, { notify: false });
```

Comment:
Use the route `contacts.detail` instead and the tests and UI will magically work.